services/weblate/weblate-user-statistic.service.js

```diff
@@ -0,0 +1,81 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+import { nonNegativeInteger, optionalUrl } from '../validators.js'
+import { metric } from '../text-formatters.js'
+
+const schema = Joi.object({
+  translated: nonNegativeInteger,
+  suggested: nonNegativeInteger,
+  uploaded: nonNegativeInteger,
+  commented: nonNegativeInteger,
+  languages: nonNegativeInteger,
+}).required()
+
+const queryParamSchema = Joi.object({
+  server: optionalUrl,
+}).required()
+
+export default class WeblateUserStatistic extends BaseJsonService {
+  static category = 'other'
+  static route = {
+    base: 'weblate',
+    pattern:
+      ':statistic(translations|suggestions|languages|uploads|comments)/:user',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: `Weblate user statistic`,
+      namedParams: { statistic: 'translations', user: 'nijel' },
+      queryParams: { server: 'https://hosted.weblate.org' },
+      staticPreview: this.render({ statistic: 'translations', count: 30585 }),
+      keywords: ['i18n', 'internationalization'],
+    },
+  ]
+
+  static defaultBadgeData = { color: 'informational' }
+
+  static render({ statistic, count }) {
+    return { label: statistic, message: metric(count) }
+  }
+
+  async fetch({ user, server = 'https://hosted.weblate.org' }) {
+    return this._requestJson({
+      schema,
+      url: `${server}/api/users/${user}/statistics/`,
+      errorMessages: {
+        403: 'access denied by remote server',
+        404: 'user not found',
+        429: 'rate limited by remote server',
+      },
+    })
+  }
+
+  async handle({ user, statistic }, { server }) {
+    let upstreamStatisticName
+
+    switch (statistic) {
+      case 'translations':
+        upstreamStatisticName = 'translated'
+        break
+      case 'suggestions':
+        upstreamStatisticName = 'suggested'
+        break
+      case 'uploads':
+        upstreamStatisticName = 'uploaded'
+        break
+      case 'comments':
+        upstreamStatisticName = 'commented'
+        break
+      default:
+        upstreamStatisticName = statistic
+    }
+
+    const data = await this.fetch({ user, server })
+    return this.constructor.render({
+      statistic,
+      count: data[upstreamStatisticName],
+    })
```

Comment:
One very common pattern you'll see in other service classes is the utilization of a `transform` in addition to the usual suspects like `handle`, `fetch`, and `render`. The `transform` function handles the responsibility of extracting the raw data values from the response and performing any translations as necessary.

It's not always used, especially in the really simple cases where a single field access is all that's needed, but this is a decent example of when extracting out the relevant portions currently here into a `transform` function would be worthwhile since the bulk of the `handle` function is now related to transformation duties.

Finally, you might find it helpful to define the route parameter mapping to response key name as an object, and then use that object to retrieve the output. This will both make the mapping a bit more succinct relative to the `switch`, especially since we're guaranteed to be exhaustive due to the bounds set on the route parameter.

E.g. something like this, which would in turn probably make it more reasonable to keep the data extraction in `handle` and not need a separate `transform` function 
```js
const statisticKeyNames = {
  translations: 'translated',
  suggestions: 'suggested',
  uploads: 'uploaded',
  comments: 'commented',
}
....
async handle({ user, statistic }, { server }) {
    const data = await this.fetch({ user, server })
    return this.constructor.render({
      statistic,
      count: data[statisticKeyNames[statistic]],
    })
}